plugins/bigquery/dbt/adapters/bigquery/connections.py

```diff
@@ -91,6 +92,25 @@ def _connection_keys(self):
         return ('method', 'database', 'schema', 'location', 'priority',
                 'timeout_seconds', 'maximum_bytes_billed')
 
+    def __post_init__(self):
+        # We need to inject the correct value of the database (aka project) at
+        # this stage, ref
+        # https://github.com/fishtown-analytics/dbt/pull/2908#discussion_r532927436.
+
+        # `database` is an alias of `project` in BigQuery
+        if self.database is None:
+            _, database = self.get_bigquery_defaults()
+            self.database = database
+
+    @classmethod
+    @lru_cache()
+    def get_bigquery_defaults(cls):
+        """ Returns (credentials, project_id) """
+        # This method is copied from ` BigQueryConnectionManager`, because it's
+        # required in both classes.
+        # We could move this & the scopes to the module level.
```

Comment:
> Generally I would have that sort of function at the module level; but put them as methods as it seems to be fairly class-focused atm. Let me know which you prefer.

Good point. @gshank @kwigley does either of you have a preference here?

Reply:
I think it makes sense to move this to module level, there's no state involved and its shared 👍 go for it!

Reply:
Cool done!

Reply:
looks like there's just a few small pep8 errors:
```
plugins/bigquery/dbt/adapters/bigquery/connections.py:48:1: E302 expected 2 blank lines, found 1
plugins/bigquery/dbt/adapters/bigquery/connections.py:50:4: W291 trailing whitespace
plugins/bigquery/dbt/adapters/bigquery/connections.py:52:1: W293 blank line contains whitespace
plugins/bigquery/dbt/adapters/bigquery/connections.py:53:52: W291 trailing whitespace
```